### DISCUSSION TITLE
Router adds operation variables in place of literals

### DISCUSSION THREAD

**Discussion by vadimofd** [wundergraph/cosmo#590] [Q&A] [answered]
It looks like the router adds new operation variables in place of literals.

This behavior seems to be undocumented, at least I wasn't able to find anything about this.

Is there any documentation on this? Or could someone help me understand the need for this? I am guessing some sort of normalization, caching, etc? Are there any limitations or issues to watch out for here?

Here is an example using Cosmo Playground and Analytics/Traces

Playground Query
```
query test_tickets($userId: GlobalID!) {
  tickets(
    TicketsFilter:{
      userId:$userId,
      ticketStatus:[ACTIVE]
    }
  ) {
    totalCount
  }
}
```

Playground Variables
```
{ "userId": "user:1" }
```

Analytics Traces Query __(notice a new `$a` variable)__
```
query test_tickets($userId: GlobalID!, $a: [TicketStatus]) {
  tickets(TicketsFilter: { userId: $userId, ticketStatus: $a }) {
    totalCount
  }
}
```

Analytics > Traces > Operation and Variables __(notice a new `$a` variable)__
```
{ "userId": "user:1", "a": ["ACTIVE"] }
```

**Comment by StarpTech** [comment, marked as answer]
Hi @vadimofd, this is expected behaviour and part of normalization.

**Reply by vadimofd** [reply]
I think it would be very helpful to have this behavior documented and probably allow for some sort of debug tracing/logging of such runtime operation changes. Also, maybe analytics tracing and playground could inform user that literals are being converted into variables?

**Reply by StarpTech** [reply]
Hi, would you be open to create a PR against our public [docs](https://github.com/wundergraph/cosmo-docs)?

**Reply by vadimofd** [reply]
I don't think I have sufficient knowledge/understanding of this normalization process and its rules to be able to document it appropriately. I've observed just a couple of scenarios in my limited experience using the product.

**Comment by dwjohnston** [comment]
@StarpTech I'm here to echo @vadimofd comments here. I've arrived at this issue with the same problem - it would be helpful to understand what the the normalization is for. I understanding it's a performance optimisation for caching.
